ngraph/core/src/op/if.cpp

```diff
@@ -0,0 +1,317 @@
+                                        ->get_results()
+                                        .at(then_desc->m_body_value_index)
+                                        ->input_value(0);
+
+            auto else_node_result = m_bodies[else_body_index]
+                                        ->get_results()
+                                        .at(else_desc->m_body_value_index)
+                                        ->input_value(0);
+
+            NODE_VALIDATION_CHECK(this,
+                                  then_node_result.get_element_type() ==
+                                      else_node_result.get_element_type(),
+                                  "type of then_body output is not equal type of else_body output");
+            
+            //shape inference for output and associated with it body outputs
+            auto partial_shape = resolve_shape(then_node_result.get_partial_shape(),
+                                               else_node_result.get_partial_shape());
+            set_output_type(output_index, then_node_result.get_element_type(), partial_shape);
+        }
+    }
+}
+
+std::shared_ptr<Node> op::v8::If::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    NGRAPH_OP_SCOPE(v8_If_clone_with_new_inputs);
+
+    check_new_args_count(this, new_args);
+    auto op = make_shared<op::v8::If>();
+    NGRAPH_CHECK(op.get(),
+                 op != nullptr,
+                 "Cannot clone ",
+                 description(),
+                 " operation with name ",
+                 get_friendly_name());
+    clone_to(*op, new_args);
+    return op;
+}
+
+op::v8::If::OutputMap op::v8::If::get_mapping_outputs_on_body_description(
+    const ngraph::op::util::MultiSubgraphOutputDescriptionVector& output_descriptors)
+{
+    OutputMap outputs_map = OutputMap();
+    std::unordered_set<int64_t> checked_results_in_body;
+
+    for (const auto& output_description : output_descriptors)
+    {
+        auto out_index = output_description->m_output_index;
+        auto internal_result_index = output_description->m_body_value_index;
+        NODE_VALIDATION_CHECK(this,
+                              checked_results_in_body.count(internal_result_index) == 0,
+                              "Incorrect associating in then_body! Result ",
+                              internal_result_index,
+                              " is already associated with another output!");
+        NODE_VALIDATION_CHECK(this,
+                              outputs_map.count(out_index) == 0,
+                              "Incorrect associating in then_body! Several results try to "
+                              "associate with the same output!");
+        checked_results_in_body.insert(internal_result_index);
+        outputs_map.insert({out_index, output_description});
+    }
+
+    return outputs_map;
+}
+void op::v8::If::clone_to(op::v8::If& dst, const OutputVector& new_args) const
```

Comment:
Why do we need to introduce the separate method? Can we do the same thing in the `clone_with_new_inputs()`?

Reply:
will be done